c++/src/kj/compat/http.c++

```diff
@@ -5974,7 +5973,7 @@ public:
     // 2. Create a promise/fulfiller pair for the status. The promise will be
     //    returned with the ConnectResponse, the fulfiller will be held by the
     //    ConnectResponseImpl.
-    auto paf = kj::newPromiseAndFulfiller<ConnectRequest::Status>();
+    auto paf = kj::newPromiseAndFulfiller<kj::Own<ConnectRequest::Status>>();
     auto forked = paf.promise.fork();
```

Comment:
It looks like only one branch is ever created off this fork? Is the fork actually needed?